include/seqan3/search/algorithm/detail/search.hpp

```diff
@@ -141,7 +141,7 @@ inline auto _search_single(index_t const & index, query_t const & query, config_
             // only one iterator is reported but it might contain more than one text position
             if (!internal_hits.empty())
             {
-                auto const & text_pos = internal_hits[0].lazy_locate();
+                auto && text_pos = internal_hits[0].lazy_locate();
```

Comment:
what does lazy_locate return and why are you taking it by `&&`?

Beware that `auto &&` only extends life-time of prvalues, not of xvalues.

Reply:
just `auto`